easybuild/easyblocks/s/scipy.py

```diff
@@ -30,34 +30,166 @@
 
     def configure_step(self):
         """Custom configure step for scipy: set extra installation options when needed."""
-        super(EB_scipy, self).configure_step()
+        pylibdir = det_pylibdir()
+
+        # scipy >= 1.9.0 uses Meson/Ninja
+        if self.use_meson:
+            # configure BLAS/LAPACK library to use with Meson for scipy >= 1.9.0
+            lapack_lib = self.toolchain.lapack_family()
+            if lapack_lib == toolchain.FLEXIBLAS:
+                blas_lapack = 'flexiblas'
+            elif lapack_lib == toolchain.INTELMKL:
+                blas_lapack = 'mkl'
+            elif lapack_lib == toolchain.OPENBLAS:
+                blas_lapack = 'openblas'
+            else:
+                raise EasyBuildError("Unknown BLAS/LAPACK library used: %s", lapack_lib)
+
+            configopts = '-Dblas=%(blas_lapack)s -Dlapack=%(blas_lapack)s' % {'blas_lapack': blas_lapack}
+            self.cfg.update('configopts', configopts)
+
+            # need to have already installed extensions in PATH, PYTHONPATH for configure/build/install steps
+            pythonpath = os.getenv('PYTHONPATH')
+            env.setvar('PYTHONPATH', os.pathsep.join([os.path.join(self.installdir, pylibdir), pythonpath]))
+
+            path = os.getenv('PATH')
+            env.setvar('PATH', os.pathsep.join([os.path.join(self.installdir, 'bin'), path]))
+
+            MesonNinja.configure_step(self)
+
+        else:
+            super(EB_scipy, self).configure_step()
 
         if LooseVersion(self.version) >= LooseVersion('0.13'):
             # in recent scipy versions, additional compilation is done in the install step,
             # which requires unsetting $LDFLAGS
             if self.toolchain.comp_family() in [toolchain.GCC, toolchain.CLANGGCC]:  # @UndefinedVariable
                 self.cfg.update('preinstallopts', "unset LDFLAGS && ")
 
+    def build_step(self):
+        """Custom build step for scipy: use ninja for scipy >= 1.9.0"""
+        if self.use_meson:
+            MesonNinja.build_step(self)
+        else:
+            super(EB_scipy, self).build_step()
+
+    def test_step(self):
+        """Run available scipy unit tests. Adapted from numpy easyblock"""
+
+        if self.use_meson:
+
+            # temporarily install scipy so we can run the test suite
+            tmpdir = tempfile.mkdtemp()
+            cwd = os.getcwd()
+
+            cmd = " && ".join([
+                # reconfigure meson to use temp dir
+                'meson configure --prefix=%s' % tmpdir,
+                'ninja -j %s install' % self.cfg['parallel'],
+            ])
+            run_cmd(cmd, log_all=True, simple=True, verbose=False)
+
+            abs_pylibdirs = [os.path.join(tmpdir, pylibdir) for pylibdir in self.all_pylibdirs]
+            for pylibdir in abs_pylibdirs:
+                mkdir(pylibdir, parents=True)
+            python_root = get_software_root('Python')
+            python_bin = os.path.join(python_root, 'bin', 'python')
```

Comment:
should use `self.python_cmd` here, which can be set by `self.prepare_python()` (defined in `PythonPackage`)